protocols/mdns/src/behaviour/iface.rs

```diff
@@ -319,10 +316,9 @@ where
                 Poll::Ready(Ok(Err(err))) => {
                     tracing::debug!("Parsing mdns packet failed: {:?}", err);
```

Comment:
I think we need to `continue` here too!

Reply:
If we hit line `317`, the socket is broken so we should probably add `return Poll::Ready(())`. If we add a `continue`, we will run into an endless loop and in the current state, it is just a memory-leak because the task stays around without ever being `poll`-ed again.

Reply:
Before I commit the changes about `continue`, I just want to be sure about the line where the socket is broken (because depending on if you talk about pre or post changes, line `317` is not the same :wink: )

For which pattern should I return `Poll::Ready(())` ?
- `Poll::Ready(Err(err)) if err.kind() == std::io::ErrorKind::WouldBlock`
- `Poll::Ready(Err(err))` 
First one ? Second one ? Or both ?

Reply:
Just the 2nd one. `WouldBlock` is what where we should return `Poll::Pending` (which happens by falling through to the end).

Reply:
Ok. But for `WouldBlock` it has now a `continue` to make sure the `waker` will be scheduled to woke back again.

If we don't put a `continue` after `WouldBlock` and go strait at the end to the `Poll::Pending`, the `waker` is not scheduled anymore (since `Poll::Ready` has just been received) and `poll` will never be called again.

Reply:
Yeah you are correct, we need the `continue` :)